### PR TITLE
Bug Fix -  Error gets thrown when supplying custom tls config

### PR DESCRIPTION
**Pull request by sean1588**
This PR puts in a bug fix to allow specifying a custom TLS config. The issue here is a call is being made to mysql.ParseDSN to parse the DSN connection string into a config object. This tool expects a value for the tls query parameter that needs to be something other than "true", "false", "skip-verify", or "preferred", in order for it to use the custom configuration we are providing it. The problem occurs because when the [call to mysql.ParseDSN](https://github.com/pulumi/golang-migrate/blob/9c15b1e30281b8a791d6e4d253e0f5535b2cc0c8/database/mysql/mysql.go#L159) is made it expects the value being passed to tls to be the name of a config that is registered. As part of parsing it it tries to look it up against registered keys, if it doesn't exist, an [error gets thrown saying "invalid / unknown tls config" ](https://github.com/go-sql-driver/mysql/blob/master/dsn.go#L132)ad the code cannot proceeed any further. At this point in time wheThis error gets th ?n the call gets made it is impossible that the config would exist as it doesn't get registered until further downnin the code execution. So the fix here is to pre-register the config if it is not one of the default values, e.g. true, flase, skip-verify, so it doesnt error out when parsing the DSN string. We then [register it "for real" further down the code path](https://github.com/pulumi/golang-migrate/blob/9c15b1e30281b8a791d6e4d253e0f5535b2cc0c8/database/mysql/mysql.go#L216). This does look a bit hacky on the surface, i.e. why not just parse the config after it is registered?, but we need to parse the config to begin with to get a hold of the values in order to register a TLS config to begin with, so we are in somewhat of a chicken-egg scenario in the way the code is implemented here, so I figured this was the safest work around.

### CODE REVIEW

File: Makefile
Comment (praneetloke) [inline Makefile:17]: I am assuming we are planning to contribute this fix upstream and delete our fork eventually? If so, any reason to remove these build instructions for the various platforms that the upstream will be expecting to build?

File: database/mysql/mysql.go
Comment (praneetloke) [inline database/mysql/mysql.go:19]: Related to my previous comment about the upstream repo. I think we should maintain their code style here, so I would suggest to revert this. But only if it looks like the other files are following a specific format.

**Review by chrsmith — APPROVED** [review]
As a quick fix to unblock Pulumi development, this seems like a fine change. So go ahead and run with it.

But if the goal is to have this PR, or something like it, try and be merged upstream, then there are a lot of ways we can improve the code and/or make it more amenable to the existing `golang-migrate/migrate` codebase.

If we were looking to do that, I'd suggest a bunch of changes. (To the point that it might make sense to just start fresh from a new feature branch.) Happy to go over some of the things that stand out in my mind over a VC.

File: database/mysql/mysql.go
Comment (praneetloke) [inline database/mysql/mysql.go:151]: What do you think about returning a custom error type rather than a wrapped-error? Consumers could then check if the error returned is of a certain type, eg. `ErrExtractTLSParam`. Note that this tool could be used programmatically as well, so it could be useful to have a defined error. Of course, if/when we contribute this fix, I have no idea what the upstream maintainers might prefer. 🤷‍♂️ 

But using the custom error type would also eliminate the need to add a new import for `github.com/pkg/errors` and thus further reduce the drift from upstream.

**Review by praneetloke — COMMENTED** [review]
The fix itself looks good to me. I didn't check how this fits into the rest of the code, though. But since this works for us, we could just let the upstream maintainers comment on if there is a better way with all their knowledge about this codebase. If you haven't already done so, could you please open an issue with upstream and submit a PR there too from our fork?

**Comment by sean1588** [comment]
> If we were looking to do that, I'd suggest a bunch of changes. (To the point that it might make sense to just start fresh from a new feature branch.) Happy to go over some of the things that stand out in my mind over a VC.

I agree there are many ways we can improve this. This was more of a quick fix to get us through, which is why I have not opened a PR against the upstream yet. But yes I would love to go over your thoughts over a VC.

File: Makefile
Comment (sean1588) [reply Makefile:17]: Yes the idea is to eventally contribute a fix upstream. Since we are going to be using this fork for now I removed the other builds becuase we are calling `make build-cli` from the dockerfile which really slows up our docker build so I got rid of them since this is only our fork and building for windows and macos are not needed.

File: database/mysql/mysql.go
Comment (sean1588) [reply database/mysql/mysql.go:151]: I think that would be ideal. For now this will unblock us though, but yea there are a lot of improvements we can make when we go to fix the upstream which is why I haven't opened a PR against the upstream yet, because there are definitely improvements we can make.

**Comment by sean1588** [comment]
I am going to go ahead and merge this to unbnlock us, but yes I think there are some improvements that need to be made before we submit an upstream fix like what @chrsmith  and @praneetloke  called out.

**Pull request merged**
